lib/zod/schema.ts

```diff
@@ -1,77 +1,263 @@
+  invalid_type_error: 'Role must be a string',
+});
+
+const sentViaEmail = z
+  .boolean({
+    invalid_type_error: 'Sent via email must be a boolean',
+  })
+  .default(false);
+
+const domains = z
+  .string({
+    invalid_type_error: 'Domains must be a string',
+  })
+  .optional()
+  .refine(
+    (domains) => (domains ? domains.split(',').every(isValidDomain) : true),
+    'Invalid domain in the list'
+  );
+
+const expiredToken = z
+  .string({
+    required_error: 'Expired token is required',
+    invalid_type_error: 'Expired token must be a string',
+  })
+  .min(1, 'Expired token is required');
+
+const sessionId = z
+  .string({
+    required_error: 'Session id is required',
+    invalid_type_error: 'Session id must be a string',
+  })
+  .min(1, 'Session id is required');
+
+const priceId = z
+  .string({
+    required_error: 'Price Id is required',
+    invalid_type_error: 'Price Id must be a string',
+  })
+  .min(1, 'PriceId is required');
+
+const quantity = z.number({
+  invalid_type_error: 'Quantity must be a number',
+});
+
+const recaptchaToken = z.string({
+  invalid_type_error: 'Recaptcha token must be a string',
+});
 
 export const createApiKeySchema = z.object({
-  name: z.string().min(1, 'Name is required').max(maxLengthPolicies.apiKeyName),
+  name: teamName,
 });
 
 export const deleteApiKeySchema = z.object({
-  apiKeyId: z.string(),
+  apiKeyId,
 });
 
 export const teamSlugSchema = z.object({
-  slug: z.string().max(maxLengthPolicies.slug),
+  slug,
 });
 
 export const updateTeamSchema = z.object({
-  name: z.string().min(1, 'Name is required').max(maxLengthPolicies.team),
-  slug: z
-    .string()
-    .min(3, 'Slug must be at least 3 characters')
-    .max(maxLengthPolicies.slug)
-    .transform((slug) => slugify(slug)),
-  domain: z
-    .string()
-    .max(maxLengthPolicies.domain)
-    .optional()
-    .refine(
-      (domain) => {
-        if (!domain) {
-          return true;
-        }
-
-        return isValidDomain(domain);
-      },
-      {
-        message: 'Enter a domain name in the format example.com',
-      }
-    )
-    .transform((domain) => {
-      if (!domain) {
-        return null;
-      }
-
-      return domain.trim().toLowerCase();
-    }),
+  name: teamName,
+  slug: slug.transform((slug) => slugify(slug)),
+  domain,
 });
 
 export const createTeamSchema = z.object({
-  name: z.string().min(1, 'Name is required').max(maxLengthPolicies.team),
+  name: teamName,
 });
 
 export const updateAccountSchema = z.union([
   z.object({
-    email: z
-      .string()
-      .email('Enter a valid email address')
-      .max(maxLengthPolicies.email),
+    email,
+  }),
+  z.object({
+    name,
   }),
   z.object({
-    name: z.string().min(1, 'Name is required').max(maxLengthPolicies.name),
+    image,
+  }),
+]);
+
+export const updatePasswordSchema = z.object({
+  currentPassword: password,
+  newPassword: password,
+});
+
+export const userJoinSchema = z.union([
```

Comment:
Check this handles all the cases:

- name, teamName, email, password (Standard signup)
- name, email, password, inviteToken (When join via invite link)
- name, password, inviteToken (When join via email invite)

Reply:
Yes, tested all 3 worked for all cases